src/xcode/ENA/ENA/Source/Scenes/Home/HomeViewController.swift

```diff
@@ -19,10 +19,18 @@ import ExposureNotification
 import UIKit
 
 protocol HomeViewControllerDelegate: AnyObject {
-	func homeViewControllerUserDidRequestReset(_ controller: HomeViewController)
+	func showRiskLegend()
+	func showExposureNotificationSetting(enState: ENStateHandler.State)
+	func showExposureDetection(state: HomeInteractor.State, isRequestRiskRunning: Bool)
+	func setExposureDetectionState(state: HomeInteractor.State, isRequestRiskRunning: Bool)
```

Comment:
why does the ExposureDetectionState has to be set in this particular way? I thought that the coordinator would only do the navigation logic. But now it is also doing side-effects by setting certain states.

Reply:
Coordinator is not only navigating, but _coordinating_ communication between different view controllers.
Coordinator can set state by design. It provides dependencies to view controllers and those dependencies usually control state and behaviour of view controller. For example, in model driven design, model contains state, and it's provided to the view controller by coordinator. 

In this case we have a scenario of two view controllers that need to share the same state. Coordinator is the right object to communicate that state between them IMO.